docs/docsite/rst/dev_guide/migrating_roles.rst

```diff
@@ -0,0 +1,345 @@
+
+This section walks through an example creating this coexistence in a downstream RPM and requires Ansible 2.9.0 or later.
+
+To deliver a role as both a standalone role and a collection role:
+
+#. Place the collection in  :file:`/usr/share/ansible/collections/ansible_collections/`.
+#. Copy the contents of the role inside the collection into a directory named after the standalone role and place the standalone role in  :file:`/usr/share/ansible/roles/`.
+
+All previously bundled modules and plugins used in the standalone role are now referenced by :abbr:`FQCN (Fully Qualified Collection Name)` so even though they are no longer embedded, they can be found from the collection contents.This is an example of how the content inside the collection is a unique entity and does not have to be bound to a role or otherwise. You could alternately create two separate collections: one for the modules and plugins and another for the standalone role to migrate to. The role must use the modules and plugins as :abbr:`FQCN (Fully Qualified Collection Name)`.
+
+The following is an example RPM spec file that accomplishes this using this example content:
+
+.. code-block:: text
+
+  Name: acme-ansible-content
+  Summary: Ansible Collection for deploying and configuring ACME webapp
+  Version: 1.0.0
+  Release: 1%{?dist}
+  License: GPLv3+
+  Source0: amce-webserver-1.0.0.tar.gz
+
+  Url: https://github.com/acme/webserver-ansible-collection
+  BuildArch: noarch
+
+  %global roleprefix my-standalone-role.
+  %global collection_namespace acme
+  %global collection_name webserver
+
+  %global collection_dir %{_datadir}/ansible/collections/ansible_collections/%{collection_namespace}/%{collection_name}
+
+  %description
+  Ansible Collection and standalone role (for backward compatibility and migration) to deploy, configure, and manage the ACME webapp software.
+
+  %prep
+  %setup -qc
+
+  %build
+
+  %install
+
+  mkdir -p %{buildroot}/%{collection_dir}
+  cp -r ./* %{buildroot}/%{collection_dir}/
+
+  mkdir -p %{buildroot}/%{_datadir}/ansible/roles
+  for role in %{buildroot}/%{collection_dir}/roles/*
+    do
+	   cp -pR ${role} %{buildroot}/%{_datadir}/ansible/roles/%{roleprefix}$(basename ${role})
+
+	   mkdir -p %{buildroot}/%{_pkgdocdir}/$(basename ${role})
+	   for docfile in README.md COPYING LICENSE
+	    do
+      	if [ -f ${role}/${docfile} ]
+    	    then
+          	cp -p ${role}/${docfile} %{buildroot}/%{_pkgdocdir}/$(basename ${role})/${docfile}
+      	fi
+	   done
+  done
+
+
+  %files
+  %dir %{_datadir}/ansible
+  %dir %{_datadir}/ansible/roles
+  %dir %{_datadir}/ansible/collections
+  %dir %{_datadir}/ansible/collections/ansible_collections
+  %{_datadir}/ansible/roles/
+  %doc %{_pkgdocdir}/*/README.md
+  %doc %{_datadir}/ansible/roles/%{roleprefix}*/README.md
+  %{collection_dir}
+  %doc %{collection_dir}/roles/*/README.md
+  %license %{_pkgdocdir}/*/COPYING
+  %license %{_pkgdocdir}/*/LICENSE
```

Comment:
https://github.com/ansible-community/collection_migration/blob/master/migrate.md has several missing steps, specially for 'non modules'

Reply:
I don't know what you are asking for in this comment?

Reply:
ok found it and moved some of them over. thanks!